net/unbound/files/unbound.init

```diff
@@ -1,20 +1,718 @@
 #!/bin/sh /etc/rc.common
-# Copyright (C) 2016 Michael Hanselmann
+#
+##############################################################################
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+#   Copyright (C) 2016 Eric Luehrsen
+#
+##############################################################################
 
-START=61
+. /lib/functions/network.sh
 
+##############################################################################
+
+START=60
 USE_PROCD=1
+PROG="/usr/sbin/unbound"
+
+##############################################################################
+
+UNBOUND_B_CONTROL=0
+UNBOUND_B_DNSMASQ=0
+UNBOUND_B_DNSSEC=0
+UNBOUND_B_GATE_NAME=0
+UNBOUND_B_LOCL_BLCK=0
+UNBOUND_B_LOCL_NAME=0
+UNBOUND_B_LOCL_SERV=1
+UNBOUND_B_OBSCURE=0
+UNBOUND_B_PRIV_BLCK=1
+
+UNBOUND_D_RESOURCE=small
+UNBOUND_D_RECURSION=passive
+
+UNBOUND_TXT_FWD_ZONE=""
+UNBOUND_TTL_MIN=120
+
+UNBOUND_N_EDNS_SIZE=1280
+UNBOUND_N_FWD_PORTS=""
+UNBOUND_N_REC_PORT=53
+UNBOUND_N_ROOT_AGE=90
+
+UNBOUND_PID="/var/run/unbound.pid"
+UNBOUND_CONF="/var/etc/unbound.conf"
+UNBOUND_HEADER_CONF="/etc/unbound/header.conf"
+
+##############################################################################
+
+domain_ptr_ip6() {
+  # Get the nibble rounded /CDIR ip6.arpa.
+  echo "$1" | awk -F: \
+  'BEGIN { OFS = "" ; }
+  { cdir = $0 ; 
+  sub(/.*\//,"",cdir) ; 
+  cdir = (cdir / 4) ;
+  sub(/\/[0-9]+/,"",$0) ; 
+  ct_stop = 9 - NF ;
+  for(i=1; i<=NF; i++) {
+    if(length($i) == 0) { 
+      for(j=1; j<=ct_stop; j++) { $i = ($i "0000") ; } } 
+    else { $i = substr(("0000" $i), length($i)+5-4) ; } } ;
+  y = $0 ;
+  ct_start = length(y) - 32 + cdir ; 
+  for(i=ct_start; i>0; i--) { x = (x substr(y,i,1)) ; } ; 
+  gsub(/./,"&\.",x) ; 
+  x = (x "ip6.arpa") ; 
+  print x }'
+}
+
+##############################################################################
+
+host_ptr_ip6() {
+  # Get host ip6.arpa.
+  echo "$1" | awk -F: \
+  'BEGIN { OFS = "" ; }
+  { sub(/\/[0-9]+/,"",$0) ; 
+  ct_stop = 9 - NF ; 
+  for(i=1; i<=NF; i++) {
+    if(length($i) == 0) { 
+      for(j=1; j<=ct_stop; j++) { $i = ($i "0000") ; } } 
+    else { $i = substr(("0000" $i), length($i)+5-4) ; } } ; 
+  y = $0 ;
+  ct_start = length(y); 
+  for(i=ct_start; i>0; i--) { x = (x substr(y,i,1)) ; } ; 
+  sub(/[0-9]+\//,"",x) ;
+  gsub(/./,"&\.",x) ; 
+  x = (x "ip6.arpa") ; 
+  print x }'
+}
+
+##############################################################################
+
+domain_ptr_ip4() { 
+  # Get the byte rounded /CDIR in-addr.arpa.
+  echo "$1" | awk \
+  '{ cdir = $0 ; 
+  sub(/.*\//,"",cdir) ; 
+  cdir = (cdir / 8) ;
+  dtxt = $0 ; 
+  sub(/\/.*/,"",dtxt) ; 
+  split(dtxt, dtxt, ".") ;
+  for(i=1; i<=cdir; i++) { x = (dtxt[i] "." x) ; }
+  x = (x "in-addr.arpa") ;
+  print x }'
+} 
 
-start_service() {
-	find /etc/unbound \! \( -user unbound -group unbound \) \
```

Comment:
You don't seem to take over this code. Someone may have unbound installed from before it used its own user (i.e. OpenWrt 15.05) and it will not work.